lib/matplotlib/animation.py

```diff
@@ -1079,7 +1079,11 @@ def func(current_frame: int, total_frames: int) -> Any
 
         def _pre_composite_to_white(color):
             r, g, b, a = mcolors.to_rgba(color)
-            return a * np.array([r, g, b]) + 1 - a
+            if type(writer) == PillowWriter:
+                print("no true transparency using pillow")
+                return a * np.array([r, g, b]) + 1 - a
+            else:
+                return np.array([r, g, b, a])
 
         savefig_kwargs['facecolor'] = _pre_composite_to_white(facecolor)
```

Comment:
Given the name of the internal function `_pre_composite_to_white`, I think it makes more sense to put the conditional around this bit of code (and just not call the function if we don't need it) rather than make the function actually do "maybe pre-composite with white based on the type of something in my outer scope".